sequencer/finalizer.go

```diff
@@ -396,32 +574,71 @@ func (f *finalizer) processTransaction(ctx context.Context, tx *TxTracker) (errW
 }
 
 // handleProcessTransactionResponse handles the response of transaction processing.
-func (f *finalizer) handleProcessTransactionResponse(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse, oldStateRoot common.Hash) (errWg *sync.WaitGroup, err error) {
+
+func (f *finalizer) handleProcessTransactionResponse(ctx context.Context, tx *TxTracker, result *state.ProcessBatchResponse, oldStateRoot common.Hash) (deviationCheckWg *sync.WaitGroup, errWg *sync.WaitGroup, err error) {
 	// Handle Transaction Error
 
 	errorCode := executor.RomErrorCode(result.Responses[0].RomError)
 	if !state.IsStateRootChanged(errorCode) {
 		// If intrinsic error or OOC error, we skip adding the transaction to the batch
 		errWg = f.handleProcessTransactionError(ctx, result, tx)
-		return errWg, result.Responses[0].RomError
+		return nil, errWg, result.Responses[0].RomError
 	}
 
 	// Check remaining resources
 	err = f.checkRemainingResources(result, tx)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	if f.cfg.EffectiveGasPrice.IsEnabled {
```

Comment:
But in the specification it says to compare it with the one form the pre-execution: https://hackmd.io/7m6an5jtRqGAaObt4EIJOw

Reply:
It says that the "attack" protection si "after second execution", not the first, as the first time it could be normal to have a difference with the estimated gasUsed.

_**After second execution**, compare again the execution and if there is still a huge difference log an event
Maybe keep the most expensive execution. (We have to decide what to do, for the moment, do nothing, just log the_ event)